File: 2914-minimum-number-of-changes-to-make-binary-string-beautiful/2914-minimum-number-of-changes-to-make-binary-string-beautiful.java
class Solution {
    public int minChanges(final String s) {
        int res = 0;
        final int n = s.length();

        for (int i = 1; i < n; i += 2) {
            if (s.charAt(i - 1) != s.charAt(i)) {
                res++;
            }
        }
        return res;
    }
}

File: 2914-minimum-number-of-changes-to-make-binary-string-beautiful/2914-minimum-number-of-changes-to-make-binary-string-beautiful.rs
impl Solution {
    pub fn min_changes(s: String) -> i32 {
        let mut c = s.chars();
        let mut res: i32 = 0;
        while let Some(x) = c.next() {
            if let Some(y) = c.next() {
                if x != y {
                    res += 1;
                }
            }
        }
        res
    }
}
